Deal with "replace-not-merge" in json-schema-validation new lib in Helm v3.18.5

benoittgt opened the pull request:
Sorry for the noise. This is a new try for https://github.com/istio/istio/pull/57445

- Helm use ssanthosh-tekuri/jsonschema v6. For draft-07, when an object contains $ref **it seems** [that siblings are ignored](https://github.com/santhosh-tekuri/jsonschema/blob/32784e992ee32ae73891b165b1319dd0bca87437/root.go#L101)(“All other properties in a ‘$ref’ object MUST be ignored”). https://github.com/santhosh-tekuri/jsonschema/issues/221#issuecomment-2861973336
    > All other properties in a "$ref" object MUST be ignored. https://json-schema.org/draft-07/draft-handrews-json-schema-01#rfc.section.8.3
- Having properties or additionalProperties at the root alongside $ref means those siblings won’t be applied. With `additionalProperties: false` in the referenced schema, unknown keys (like `_internal_defaults_do_not_set`) are rejected.


I think that [@keithmattix is right](https://github.com/istio/istio/pull/57445#issuecomment-3212284177)
> Best I can tell, if you set properties at root, the subschema (set by $ref) no longer takes effect so you have to declare the entire schema in properties.

And I see this only way to fix the Chart now.

Before:
```
helm lint .
==> Linting .
[ERROR] values.yaml: - at '': additional properties '_internal_defaults_do_not_set' not allowed

[ERROR] templates/: values don't meet the specifications of the schema(s) in the following chart(s):
gateway:
- at '': additional properties '_internal_defaults_do_not_set' not allowed
```

After:
```
helm lint .
==> Linting .

1 chart(s) linted, 0 chart(s) failed
```

Related:

- https://github.com/istio/istio/pull/57426
- https://github.com/helm/helm/issues/31148
- https://github.com/istio/istio/issues/57354

### Minimal way to reproduce

`value.schema.json`
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$defs": {
    "values": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "name": { "type": "string" }
      }
    }
  },
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "_internal_defaults_do_not_set": { "type": "object" }
  },
  "$ref": "#/$defs/values"
}
```

`values.yaml`
```yaml
_internal_defaults_do_not_set: {}
name: "demo"
```
and.. the `Chart.yaml`
```yaml
apiVersion: v2
name: schema-demo-bad
version: 0.1.0
```

It will work with
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$defs": {
    "values": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "_internal_defaults_do_not_set": { "type": "object" },
        "name": { "type": "string" }
      }
    }
  },
  "$ref": "#/$defs/values"
}
```
Because of $ref replacement.

benoittgt committed:
Fix schema to allow _internal_defaults_do_not_set without breaking $ref

- Keeps a single source of truth for top-level keys in "$defs.values.properties".
- Helm uses santhosh-tekuri/jsonschema v6. For draft-07, when an object contains
$ref it seems siblings are ignored. https://github.com/santhosh-tekuri/jsonschema/blob/32784e992ee32ae73891b165b1319dd0bca87437/root.go#L101
- Having properties or additionalProperties at the root alongside $ref
means those siblings won’t be applied. With additionalProperties: false
in the referenced schema, unknown keys (like
_internal_defaults_do_not_set) are rejected.

Before:
```
helm lint .
==> Linting .
[ERROR] templates/: template: gateway/templates/service.yaml:1:22: executing "gateway/templates/service.yaml" at <.Values.service.type>: nil pointer evaluating interface {}.type

Error: 1 chart(s) linted, 1 chart(s) failed
```

After:
```
helm lint .
==> Linting .

1 chart(s) linted, 0 chart(s) failed
```

Related:

- https://github.com/istio/istio/pull/57426
- https://github.com/helm/helm/issues/31148
- https://github.com/istio/istio/issues/57354

Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

istio-testing commented:
Hi @benoittgt. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

keithmattix commented:
/ok-to-test

benoittgt committed:
Add 3 missing fields in schema

Avoid
```
--- FAIL: TestRender (0.00s)
    --- FAIL: TestRender/gateway-additional-containers (0.00s)
        helm_test.go:148:
            	Error Trace:	/home/prow/go/src/istio.io/istio/operator/pkg/helm/helm_test.go:148
            	Error:      	Received unexpected error:
            	            	render chart: converting values: values don't meet the specifications of the schema(s) in the following chart(s):
            	            	gateway:
            	            	- at '': additional properties 'additionalContainers' not allowed
            	Test:       	TestRender/gateway-additional-containers
    --- FAIL: TestRender/gateway-init-containers (0.00s)
        helm_test.go:148:
            	Error Trace:	/home/prow/go/src/istio.io/istio/operator/pkg/helm/helm_test.go:148
            	Error:      	Received unexpected error:
            	            	render chart: converting values: values don't meet the specifications of the schema(s) in the following chart(s):
            	            	gateway:
            	            	- at '': additional properties 'initContainers' not allowed
            	Test:       	TestRender/gateway-init-containers
    --- FAIL: TestRender/gateway-env-var-from (0.00s)
        helm_test.go:148:
            	Error Trace:	/home/prow/go/src/istio.io/istio/operator/pkg/helm/helm_test.go:148
            	Error:      	Received unexpected error:
            	            	render chart: converting values: values don't meet the specifications of the schema(s) in the following chart(s):
            	            	gateway:
            	            	- at '': additional properties 'envVarFrom' not allowed
            	Test:       	TestRender/gateway-env-var-from
```

https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/gencheck_istio/1959012558157385728
Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 951f5970ca7afbda58f0ea1072e59eb96c7e1839 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/gencheck_istio/1959012558157385728) | true | `/test gencheck`
unit-tests-arm64_istio | 951f5970ca7afbda58f0ea1072e59eb96c7e1839 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/unit-tests-arm64_istio/1959012572556431360) | true | `/test unit-tests-arm64`
release-notes_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959014705141911552) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959014705141911552) | true | `/test release-notes`
gencheck_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/gencheck_istio/1959014689136447488) | true | `/test gencheck`
integ-helm_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959014690403127296) | true | `/test integ-helm`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

benoittgt commented:
/retest

It should be good as I am not able to reproduce them with the last commit locally with `go test ./operator/pkg/helm -v`

keithmattix commented:
Ah looks like it does affect our integration tests though. Seems like there's still some failures somewhere (maybe in our tests themselves)

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959017124206743552) | true | `/test release-notes`
gencheck_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/gencheck_istio/1959017123980251136) | true | `/test gencheck`
integ-helm_istio | 47e61fbb7eb3df1dec88cce95311070fb471ed5c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959017124093497344) | true | `/test integ-helm`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

benoittgt committed:
Add missing `make gen`

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 270d6a3a9a4c4c4eb101d549c5f80b7b15f18f4e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959607544846487552) | true | `/test release-notes`
integ-helm_istio | 270d6a3a9a4c4c4eb101d549c5f80b7b15f18f4e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959607532456513536) | true | `/test integ-helm`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

benoittgt committed:
Fix "Additional property pilot is not allowed"

When doing:

`go test -tags=integ -timeout 30s -run ^TestAmbientInstallMultiNamespace$ -v ./tests/integration/helm`

Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-helm_istio | 270d6a3a9a4c4c4eb101d549c5f80b7b15f18f4e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959607532456513536) | true | `/test integ-helm`
release-notes_istio | bc608ba5f7ebc376a2138342333986ab392531d9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959744334416842752) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | bc608ba5f7ebc376a2138342333986ab392531d9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959744334416842752) | true | `/test release-notes`
integ-helm_istio | bc608ba5f7ebc376a2138342333986ab392531d9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959744321842319360) | true | `/test integ-helm`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

benoittgt committed:
Fix "Additional property defaultRevision is not allowed"

https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959744321842319360
Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

istio-testing commented:
@benoittgt: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-helm_istio | bc608ba5f7ebc376a2138342333986ab392531d9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/integ-helm_istio/1959744321842319360) | true | `/test integ-helm`
release-notes_istio | 4e0f145b73b75c6284d9a38f1618432fec1e1c19 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57457/release-notes_istio/1959749507780448256) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

benoittgt commented:
@keithmattix it seems much better now.

benoittgt committed:
Add missing release note

Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

dhawton commented on releasenotes/notes/57354.yaml:
```diff
@@ -0,0 +1,8 @@
+  - 57354
+releaseNotes:
+  - |
+    **Fixed** an issue where the Helm 3.18.5 switched to a new JSON schema validator that is more strict and caused installation failures in the Gateway chart.
```
```suggestion
    **Fixed** an issue where Helm 3.18.5 switched to a new JSON schema validator that is more strict and caused installation failures in the Gateway chart.
```

benoittgt replied:
Thanks. I made this https://github.com/istio/istio/pull/57457/commits/dd3141590b4ce0daba6849581a5acb66370dd88a

benoittgt committed:
Fix english in releaseNotes

Signed-off-by: Benoit Tigeot <benoit.tigeot@lifen.fr>

dhawton commented:
> Thanks this is definitely looking better! @dhawton what do you think about merging this and then trying to bump helm in build tools again?

To maybe not block master, I think we merge this and maybe locally build a build-tools image with the helm bump and run some tests to make sure it's resolved.

keithmattix commented:
/cherry-pick release-1.26

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.27":
```
Applying: Fix schema to allow _internal_defaults_do_not_set without breaking $ref
Using index info to reconstruct a base tree...
M	go.mod
M	go.sum
M	manifests/charts/gateway/values.schema.json
Falling back to patching base and 3-way merge...
Auto-merging manifests/charts/gateway/values.schema.json
Auto-merging go.sum
Auto-merging go.mod
CONFLICT (content): Merge conflict in go.mod
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix schema to allow _internal_defaults_do_not_set without breaking $ref

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236243):

>/cherry-pick release-1.27


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57508

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236243):

>/cherry-pick release-1.27


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.26":
```
Applying: Fix schema to allow _internal_defaults_do_not_set without breaking $ref
Using index info to reconstruct a base tree...
M	go.mod
M	go.sum
M	manifests/charts/gateway/values.schema.json
Falling back to patching base and 3-way merge...
Auto-merging manifests/charts/gateway/values.schema.json
Auto-merging go.sum
Auto-merging go.mod
CONFLICT (content): Merge conflict in go.mod
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix schema to allow _internal_defaults_do_not_set without breaking $ref

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236363):

>/cherry-pick release-1.26


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57509

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236363):

>/cherry-pick release-1.26


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57510

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236500):

>/cherry-pick release-1.25


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.25":
```
Patch is empty.
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To record the empty patch as an empty commit, run "git am --allow-empty".
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236500):

>/cherry-pick release-1.25


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57738

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236500):

>/cherry-pick release-1.25


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.27":
```
Applying: Fix schema to allow _internal_defaults_do_not_set without breaking $ref
Using index info to reconstruct a base tree...
M	go.mod
M	go.sum
M	manifests/charts/gateway/values.schema.json
Falling back to patching base and 3-way merge...
Auto-merging manifests/charts/gateway/values.schema.json
Auto-merging go.sum
Auto-merging go.mod
CONFLICT (content): Merge conflict in go.mod
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix schema to allow _internal_defaults_do_not_set without breaking $ref

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236243):

>/cherry-pick release-1.27


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57740

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236243):

>/cherry-pick release-1.27


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.26":
```
Applying: Fix schema to allow _internal_defaults_do_not_set without breaking $ref
Using index info to reconstruct a base tree...
M	go.mod
M	go.sum
M	manifests/charts/gateway/values.schema.json
Falling back to patching base and 3-way merge...
Auto-merging manifests/charts/gateway/values.schema.json
Auto-merging go.sum
Auto-merging go.mod
CONFLICT (content): Merge conflict in go.mod
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix schema to allow _internal_defaults_do_not_set without breaking $ref

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236363):

>/cherry-pick release-1.26


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.25":
```
Patch is empty.
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To record the empty patch as an empty commit, run "git am --allow-empty".
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236500):

>/cherry-pick release-1.25


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: #57457 failed to apply on top of branch "release-1.26":
```
Applying: Fix schema to allow _internal_defaults_do_not_set without breaking $ref
Using index info to reconstruct a base tree...
M	go.mod
M	go.sum
M	manifests/charts/gateway/values.schema.json
Falling back to patching base and 3-way merge...
Auto-merging manifests/charts/gateway/values.schema.json
Auto-merging go.sum
Auto-merging go.mod
CONFLICT (content): Merge conflict in go.mod
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix schema to allow _internal_defaults_do_not_set without breaking $ref

```

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236363):

>/cherry-pick release-1.26


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@keithmattix: new issue created for failed cherrypick: #57745

<details>

In response to [this](https://github.com/istio/istio/pull/57457#issuecomment-3235236363):

>/cherry-pick release-1.26


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>
